src/Columns/ColumnObject.cpp

```diff
@@ -626,6 +626,9 @@ ColumnObject::ColumnObject(Subcolumns && subcolumns_, bool is_nullable_)
 
 void ColumnObject::checkConsistency() const
 {
+    if (num_rows && subcolumns.empty())
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "ColumnObject is inconsistent: it has no subcolumns, but has {} rows", num_rows);
```

Comment:
Actually this assertion is wrong because such situation is normal. Consider the following data: `{} {"key1": "value1"}`. In this case we have to insert default value with no subcolumns for first object to preserve correct number of of rows and then to fill first row of subcolumn for `key1` with default value. If there were no subcolumns at the end of processing the dummy subcolum is added because we cannot have empty tuples (see `ColumnObject::finalize`).